plugins/defaultplugins/ifplugin/interface_config.go

```diff
@@ -522,6 +542,44 @@ func (plugin *InterfaceConfigurator) modifyRxMode(ifIdx uint32, newIntf *intf.In
 	return err
 }
 
+func (plugin *InterfaceConfigurator) modifyContainerIPAddress(newConfig *intf.Interfaces_Interface, oldConfig *intf.Interfaces_Interface, ifIdx uint32) (int8, error) {
+	oldAddr, _, err := addrs.ParseIPWithPrefix(oldConfig.ContainerIpAddress)
+	if err != nil {
+		return 1, err
+	}
+	newAddr, _, err := addrs.ParseIPWithPrefix(newConfig.ContainerIpAddress)
+	if err != nil {
+		return 1, err
+	}
+	toBeDeleted, toBeAdded := addrs.DiffAddr([]*net.IPNet{newAddr}, []*net.IPNet{oldAddr})
+
+	plugin.Log.Debug("Container IP address modification (delete): ", toBeDeleted)
+	plugin.Log.Debug("Container IP address modification (add): ", toBeAdded)
+
+	var wasError error
+	for i := range toBeDeleted {
```

Comment:
what happens if you call SetContainerIP if it already exists? I mean do we need to remove the old one?
And toBeDeleted will be always just one value (or none)

Reply:
you are right in both cases